tools/sqlite3_api_wrapper/sqlite3_udf_api/sqlite3_udf_wrapper.cpp

```diff
@@ -0,0 +1,85 @@
+#include "sqlite3_udf_wrapper.hpp"
+#include "cast_sqlite.hpp"
+#include <cassert>
+
+#include "sqlite3.h"
+
+scalar_function_t SQLiteUDFWrapper::CreateSQLiteScalarFunction(scalar_sqlite_udf_t sqlite_udf, sqlite3 *db_sqlite3,
+                                                               void *pApp) {
+	scalar_function_t udf_function = [=](DataChunk &args, ExpressionState &state, Vector &result) -> void {
+		CastSQLite::InputVectorsToVarchar(args);
+
+		// Orrify all input colunms
+		unique_ptr<VectorData[]> vec_data = args.Orrify();
+
+		// Vector of sqlite3_value for all input columns
+		vector<unique_ptr<vector<sqlite3_value>>> vec_sqlite(args.ColumnCount());
+		// Casting input data to vectors of sqlite_value
+		VectorType result_vec_type = CastSQLite::ToVectorsSQLiteValue(args, result, vec_sqlite, move(vec_data));
+		result.SetVectorType(result_vec_type);
+
+		SQLiteTypeValue res_sqlite_value_type =
+		    SQLiteTypeValue::NULL_VALUE; // to hold the last sqlite value type set by UDF
+
+		sqlite3_context context;
+		context.pFunc.pUserData = pApp; // set the function data
+
+		size_t argc = args.ColumnCount();                     // num of args for the UDF
+		vector<sqlite3_value> res_sqlite_values(args.size()); // to store the results from the UDF calls
+		vector<sqlite3_value> vec_values_to_free; // sqlite_values to free because some memory allocation has happened
+
+		// traversing the vector of sqlite values
+		for (idx_t row_idx = 0; row_idx < res_sqlite_values.size(); ++row_idx) {
+
+			sqlite3_value **argv = new sqlite3_value *[argc];
```

Comment:
Could we avoid allocating this for every row iteration?
Could we wrap this in a unique_ptr, to avoid having to manually call delete later on (e.g. `unique_ptr<sqlite_value[]>(new sqlite_value[argc])`)?